common/reportMetrics.py

```diff
@@ -369,108 +373,343 @@ def process_replay(file_path):
+                    current_obj['timestamp'] = _get_timestamp_db2(timestamp_str, parameters['timeZone'])
+                    current_obj['id'] = line.split()[1]
+                    current_obj['LEVEL'] = line[line.rfind(' '):]
+                elif line[:3] == 'ARG' or line[:4] == 'DATA' or line[:7] == 'MESSAGE':
+                    key = line[:line.find(':')]
+                    current_obj[key] = line[line.find(':') + 1:]
+                    # until we hit the next field
+                    i = log_file.tell()  # save the current position
+                    next_line = log_file.readline()
+                    while next_line.strip():
+                        current_obj[key] = current_obj[key] + next_line
+                        i = log_file.tell()  # save the current position
+                        next_line = log_file.readline()
+                        if next_line[:3] == 'ARG' or next_line[:4] == 'DATA' or next_line[:7] == 'MESSAGE':
+                            break
+                    log_file.seek(i)
+                else:
+                    extract_fields_db2(current_obj, line)
+
+            # if the last loop results in chunkLines being set...
+            if line_count == parameters['chunkLines']:
+                logger.debug("--- Chunk creation time: %s seconds ---" % (time.time() - start_time))
+                send_data(current_row, log_file_path, chunk_count)
+                current_row = []
+                chunk_count += 1
+                line_count = 0
+                start_time = time.time()
+
+            line = log_file.readline()
+        # last chunk
+        if len(current_row) != 0:
+            # build json entry
+            entry = dict()
+            entry['tag'] = 'db2-replay'
+            entry['eventId'] = str(current_obj.pop('timestamp'))
+            entry['data'] = current_obj
+            current_row.append(entry)
+            logger.debug("--- Chunk creation time: %s seconds ---" % (time.time() - start_time))
+            send_data(current_row, log_file_path, chunk_count)
+            chunk_count += 1
+        logger.debug("Total chunks created: " + str(chunk_count))
+    return
+
+
+def extract_fields_db2(obj, line):
+    # need to put spaces before :
+    matches = re.findall('[A-Z]+:', line)
+    for match in matches:
+        split_at = line.find(':', line.find(match))
+        line = line[:split_at] + ' ' + line[split_at:]
+
+    line_arr = line.split()
+    while len(line_arr) > 0:
+        rev_array = line_arr[::-1]
+        last_val_pos = len(line_arr) - rev_array.index(':')
+        last_val = ' '.join(line_arr[last_val_pos:])
+        last_key_pos = last_val_pos - 2
+        last_key = line_arr[last_key_pos]
+        obj[last_key] = last_val
+        if last_key_pos == 0:
+            break
+        line_arr = line_arr[:last_key_pos]
+    return
+
+
+def replay_gpfs(log_file_path):
+    print('Replaying gpfs file')
+    with open(log_file_path) as log_file:
```

Comment:
check if file exists before opening

Reply:
added check when input as parameter, already checked at top of tag that calls this